source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -687,16 +687,18 @@ def merge_responses(responses):
 
     if not merged_response.journeys:
         # we aggregate the errors found
-
         errors = {r.error.id: r.error for r in responses if r.HasField(str('error'))}
-        if len(errors) == 1:
-            merged_response.error.id = list(errors.values())[0].id
-            merged_response.error.message = list(errors.values())[0].message
-        else:
-            # we need to merge the errors
-            merged_response.error.id = response_pb2.Error.no_solution
-            merged_response.error.message = "several errors occured: \n * {}"\
-                .format("\n * ".join([m.message for m in errors.values()]))
+
+        if len(errors) > 0:
+
+            if len(errors) == 1:
+                merged_response.error.id = list(errors.values())[0].id
+                merged_response.error.message = list(errors.values())[0].message
+            else:
+                # we need to merge the errors
+                merged_response.error.id = response_pb2.Error.no_solution
+                merged_response.error.message = "several errors occured: \n * {}"\
+                    .format("\n * ".join([m.message for m in errors.values()]))
```

Comment:
the whole bloc can be written as follows:

```python
if len(errors) == 1:
    merged_response.error.id = next(errors.values()).id   # it seems values() returns a generator, not sure, to be tested
    merged_response.error.id = response_pb2.Error.no_solution
    merged_response.error.message = next(errors.values()).message
elif len(errors) > 1:
    merged_response.error.id = response_pb2.Error.no_solution
    merged_response.error.message = "several errors occured: \n * {}"\
        .format("\n * ".join((m.message for m in errors.values())))
```

Reply:
Yes, it is a better syntax